src/consumer/runner.js

```diff
@@ -90,10 +90,25 @@ module.exports = class Runner extends EventEmitter {
     this.scheduleFetchManager()
   }
 
-  async scheduleFetchManager() {
+  scheduleFetchManager() {
+    if (!this.running) {
+      this.consuming = false
+
+      this.logger.info('consumer not running, exiting', {
+        groupId: this.consumerGroup.groupId,
+        memberId: this.consumerGroup.memberId,
+      })
+
+      return
+    }
+
     this.consuming = true
 
-    while (this.running) {
+    this.retrier(async (bail, retryCount, retryTime) => {
```

Comment:
This will now crash once the retrier runs out of attempts in case of repeated rebalances. Before, it would just continue to attempt to fetch. I can't really tell if that's a feature or a bug, but it is what I assume is an unintended change.